packages/api/src/platforms/vtex/resolvers/validateCart.ts

```diff
@@ -47,23 +51,26 @@ const groupById = (offers: IStoreOffer[]): Map<string, IStoreOffer> =>
     return acc
   }, new Map<string, IStoreOffer>())
 
-const equals = (of1: OrderForm, of2: OrderForm) => {
-  const pick = ({ orderFormId, messages, items, salesChannel }: OrderForm) => ({
-    orderFormId,
-    messages,
-    salesChannel,
-    items: items.map(
-      ({ uniqueId, quantity, seller, sellingPrice, availability }) => ({
-        uniqueId,
-        quantity,
-        seller,
-        sellingPrice,
-        availability,
-      })
-    ),
+const equals = (storeOrder: IStoreOrder, orderForm: OrderForm) => {
+  const pick = (item: Indexed<IStoreOffer>, index: number) => ({
+    ...item,
+    itemOffered: {
+      sku: item.itemOffered.sku,
+    },
+    index,
   })
 
-  return deepEquals(pick(of1), pick(of2))
+  const orderFormItems = orderForm.items.map(orderFormItemToOffer).map(pick)
+  const storeOrderItems = storeOrder.acceptedOffer.map(pick)
+
+  if (
+    storeOrder.orderNumber !== orderForm.orderFormId ||
+    !deepEquals(orderFormItems, storeOrderItems)
+  ) {
+    return false
+  }
+
+  return true
```

Comment:
What do you think about this suggestion to improve the readability? (trying to extract to good names hahah)

```suggestion
  const isSameOrder = storeOrder.orderNumber === orderForm.orderFormId
  const orderItensAreSync = deepEquals(orderFormItems, storeOrderItems)

  return isSameOrder && orderItensAreSync
```